File: easy/age-distribution.py
#"""Sample code to read in test cases:

import sys
test_cases = open(sys.argv[1], 'r')
human='This program is for humans'
res=['Still in Mama\'s arms'] * 3 + ['Preschool Maniac'] * 2 +  ['Elementary school'] * 7 +  ['Middle school' ] * 3 +    ['High school' ] * 4 +    ['College'] * 4 +    ['Working for the man' ] * 43 +    ['The Golden Years' ] * 35
for test in test_cases:
    # ignore test if it is an empty line
    # 'test' represents the test case, do something with it
    # ...
    # ...
    test=test.strip()
    if len(test)==0:
        continue
    i=int(test)
    print(res[i] if 0 <= i <= 100 else human)
    
test_cases.close()
#"""

File: codeeval/easy/age-distribution.cpp
/*
 * Common base C++ source for codeeval solutions.
 * */
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

void test();
void process_file(char*);
int main(int argc, char ** argv) {
//	test();
	process_file(argv[1]);
	return 0;
}

void process(std::string s);

void test() {
	process("0");
	process("19");
//	std::cout << ""  "\n";// std::endl for flush?
}

void process_file(char* path) {
	std::ifstream stream(path);
	std::string line;
	while(std::getline(stream, line)) {
		process(line);
	}
}

namespace {
    std::vector<std::string> v {
        "Still in Mama's arms",
        "Preschool Maniac",
        "Elementary school",
        "Middle school",
        "High school",
        "College",
        "Working for the man",
        "The Golden Years",
        "This program is for humans",
    };
}

void process(std::string s) {
//	std::cout << "s = '" << s << "'\n";
    auto i = std::stoi(s);
    int idx = 0;
    if(i < 0) i = 101;
    if(i > 2) ++idx;
    if(i > 4) ++idx;
    if(i > 11) ++idx;
    if(i > 14) ++idx;
    if(i > 18) ++idx;
    if(i > 22) ++idx;
    if(i > 65) ++idx;
    if(i > 100) ++idx;
    std::cout << v[idx] << "\n";
}
